File: backend/auto-trading-bot/src/main/java/com/trading/autotradingbot/dto/BotConfigDto.java
package com.trading.autotradingbot.dto;

import com.trading.autotradingbot.entity.enums.TradingMode;
import com.trading.autotradingbot.entity.enums.BotStatus;
import lombok.Data;

@Data
public class BotConfigDto {
    private String selectedSymbol;
    private TradingMode tradingMode;
    private BotStatus status;
}

File: frontend/src/types/dto/BotConfigDto.ts
// frontend/src/types/dto/BotConfigDto.ts

// Union types for enforced string values (mirroring Java Enums)
export type TradingMode = 'TRADING' | 'TRAINING';
export type BotStatus = 'RUNNING' | 'PAUSED' | 'IDLE';

export interface BotConfigDto {
    selectedSymbol: string;
    tradingMode: TradingMode;
    status: BotStatus;
}